engine/src/main/java/org/hibernate/search/engine/spi/EntityIndexBinding.java

```diff
@@ -50,6 +50,16 @@
 	void postInitialize(Set<Class<?>> indexedClasses);
 
 	/**
+	 * @return the {@code IndexManager} impl used by this entity type
+	 */
+	Class<? extends IndexManager> getIndexManagerImpl();
```

Comment:
Yes, that's what I used first.

But the fact is that with dynamic sharding strategy, it does not work as `getIndexManagers()` starts the index managers and it is too soon to create the index managers in this case. See for instance `IndexManagerHolder#buildEntityIndexBinding` where there is a special case for dynamic sharding to avoid starting the index managers.

That's why I use the class and that's why the name of the method is `getIndexManagerImpl`.